### DISCUSSION TITLE
Whether a transaction ID can be extracted from a connection object.

### DISCUSSION THREAD

**Discussion by JangKyumin** [psycopg/psycopg2#1377] [Q&A] [answered]
Hello :)
I'd like to ask you a question about transaction management.
It is premised that the query has been executed in the auto commit, connection cursor. 
Is there any way to extract the transaction ID from the connection object?
I'd appreciate it if you could answer.

**Comment by dvarrazzo** [comment, marked as answer]
The connection itself doesn't receive a transaction id you can extract from the server.

You can use a function such as [` pg_current_xact_id()`](https://www.postgresql.org/docs/current/functions-info.html#FUNCTIONS-PG-SNAPSHOT) and return it together with your result, or use a form such as `INSERT INTO.... RETURNING pg_current_xact_id()` maybe?

**Reply by JangKyumin** [reply]
Thank you for your kind answer :)
